siliconcompiler/utils/settings.py

```
@@ -37,44 +43,48 @@ def _load(self):
+                # Ensure the loaded data is actually a dictionary
+                if isinstance(data, dict):
+                    self.__settings = data
+                else:
+                    # If valid JSON but not a dict (e.g. a list), reset to empty
+                    self.__logger.warning(f"File {self.__filepath} did not contain a JSON object. "
+                                          "Resetting.")
+                    self.__settings = {}
+
+            except json.JSONDecodeError:
+                self.__logger.error(f"File {self.__filepath} is malformed. "
                                     "Starting with empty settings.")
```

Ruff (0.14.10)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)